src/Symfony/Component/Workflow/Registry.php

```diff
@@ -24,17 +23,11 @@ class Registry
     private $workflows = array();
 
     /**
-     * @param Workflow                        $workflow
-     * @param string|SupportStrategyInterface $supportStrategy
+     * @param Workflow                 $workflow
+     * @param SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $supportStrategy)
+    public function add(Workflow $workflow, SupportStrategyInterface $supportStrategy)
```

Comment:
But there was a deprecation notice in 3.* and now it's 4.0 so we can broke the BC, right ?

And I agree there is no clean upgrade path. Users should just add this new type hint.

Do you think I should remove this?